app/groups/[groupId]/details/index.tsx

```diff
@@ -0,0 +1,226 @@
+    groupDetails?.attachmentId,
+  );
+  const isOwner = authState.userId === groupDetails?.ownerId;
+
+  const dataPresentAndNoFetching =
+    groupDetails && groupPicture && !isFetchGroupPicture && !isFetchingGroupDetails;
+
+  const { groupUpdate, setGroupUpdate } = useContext(GroupUpdateContext);
+
+  useLayoutEffect(() => {
+    if (isOwner && dataPresentAndNoFetching) {
+      setGroupUpdate({ groupName: groupDetails.name, groupPicture, isValidGroupName: true });
+    }
+  }, [dataPresentAndNoFetching, isOwner]);
+
+  const [bothRunning, setBothRunning] = useState(false);
+  const {
+    mutate: updateGroupPicture,
+    isPending: isUpdatedGroupPicturePending,
+    isSuccess: isUpdatedGroupPictureSuccess,
+    isError: isUpdatedGroupPictureError,
+  } = useUpdateGroupPicture(bothRunning, params.groupId, groupDetails?.attachmentId);
+
+  const {
+    mutate: updateGroupDetails,
+    isPending: isUpdatedGroupDetailsPending,
+    isSuccess: isUpdatedGroupDetailsSuccess,
+    isError: isUpdatedGroupDetailsError,
+  } = useUpdateGroup(bothRunning, params.groupId);
+
+  useEffect(() => {
+    if (isUpdatedGroupDetailsSuccess && isUpdatedGroupPictureSuccess) {
+      router.back();
+    }
+  }, [isUpdatedGroupDetailsSuccess, isUpdatedGroupPictureSuccess]);
+
+  useEffect(() => {
+    if (isUpdatedGroupDetailsError || isUpdatedGroupPictureError) {
+      router.push(`/groups/${params.groupId}/details/(modal)/error-modal`);
+    }
+  }, [isUpdatedGroupDetailsError, isUpdatedGroupPictureError]);
+
+  const pickImage = async () => {
+    const result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [1, 1],
+      quality: 1,
+      base64: true,
+    });
+
+    if (!result.canceled) {
+      handleImageChoice(
+        result.assets[0],
+        () =>
+          router.push(`/groups/${params.groupId}/details/(modal)/unsupported-file-format-modal`),
+        () => router.push(`/groups/${params.groupId}/details/(modal)/image-too-large-modal`),
+        () =>
+          setGroupUpdate({
+            ...groupUpdate,
+            groupPicture: {
+              uri: `data:${result.assets[0].mimeType};base64,${result.assets[0].base64}`,
+            },
+          }),
+      );
+    }
+  };
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerShown: true,
+
+      header: () => (
+        <CustomHeader
+          title={t('Group data')}
+          onLeftIconPress={() => {
+            if (isOwner && dataChanged()) {
+              router.push(`/groups/${params.groupId}/details/(modal)/exit-without-saving-modal`);
+            } else {
+              router.back();
+            }
+          }}
+          isLoading={isUpdatedGroupDetailsPending || isUpdatedGroupPicturePending}
+        />
+      ),
+    });
+  }, [navigation, groupUpdate, isUpdatedGroupDetailsPending, isUpdatedGroupPicturePending]);
+
+  function handleBackClick() {
```

Comment:
in one function we specify return type in that case we also want to do that or not ?